### ISSUE TITLE
Upgrading federated servers fails with version 3.0.1

### ISSUE THREAD

**Issue by scma-esrich** [Esri/arcgis-powershell-dsc#255]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request

### Module Version

* 3.0.1

### Affected Resource(s)

* ArcGIS_Server (version-independent)

### Configuration Files

The problem doesn't lie within the JSON-config.

### Expected Behavior

A federated Image Server should have got upgraded from version 10.7.1 to version 10.8.

### Actual Behavior

Before any operation really starts, the execution stops with the following error:
```
PS C:\Temp> Invoke-ArcGISConfiguration -ConfigurationParametersFile C:\Temp\GISServer-Image-Server-108.json -Mode Upgrade -DebugSwitch
ConvertTo-SecureString : Das Argument kann nicht an den Parameter "String" gebunden werden, da es NULL ist.
In C:\Program Files\WindowsPowerShell\Modules\ArcGIS\ArcGIS.psm1:1500
Zeichen:263
+ ... ecureString
$PortalConfig.ConfigData.Portal.PortalAdministrator.Passw ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidData: (:) [ConvertTo-SecureString], 
ParameterBindingValidationException
    + FullyQualifiedErrorId : 
ParameterArgumentValidationErrorNullNotAllowed,Microsoft.PowerShell.Commands.ConvertToSecureStringCommand
 
New-Object : Ausnahme beim Aufrufen von ".ctor" mit 2 Argument(en):  "Das Argument kann nicht verarbeitet werden, da der Wert des Arguments "userName" ungültig ist. Ändern Sie den Wert des Arguments "userName", und führen Sie den Vorgang erneut aus."
In C:\Program Files\WindowsPowerShell\Modules\ArcGIS\ArcGIS.psm1:1501
Zeichen:54
+ ... redential = New-Object -TypeName
System.Management.Automation.PSCrede ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [New-Object], 
MethodInvocationException
    + FullyQualifiedErrorId : 
ConstructorInvokedThrowException,Microsoft.PowerShell.Commands.NewObjectCommand
 
Other Server Upgrade
WA Server Uninstall
Dot Sourcing the Configuration:- WebAdaptorUninstall
...
```
After that, the process was killed by the user.

### Steps to Reproduce

Start an upgrade of a federated server with the current sample-config and the DSC-version 3.0.1.

### Important Factoids

The problem lies within ArcGIS.psm1, rows 1500 and 1501:
- The module tries to get the PortalAdministrator-password ouf of the $PortalConfig based on $cfHashtable.
- It is looking at the location $PortalConfig.ConfigData.**Portal**.PortalAdministrator.UserName and $PortalConfig.ConfigData.**Portal**.PortalAdministrator.Password. However, in a federated server-upgrade there is no "Portal"-block in the $cfHashtable.

**Workarounds**
- Changing "Portal" to "Federation" in ArcGIS.psm1,rows 1500 and 1501 on the push-host did solve the issue in this specific situation but will potentially break the upgrade of a base-deployment.
- Another workaround would be to add a "Portal"-block to the JSON-config, containing the PortalAdministrator-object.

**Suggestion for a fix**
In my opinion, a valuable solution would be to integrate a check similar to ArcGIS.psm1, rows 1436 and 1437 separating those two scenarios.

**Comment by scma-esrich** [comment]
@trffrm, @spitzerr, @esride-dku, @jesa-esrich - FYI.

**Comment by cameronkroeker** [comment]
Addressed in v3.0.2.

**Comment by scma-esrich** [comment]
Thanks!
